### PR TITLE
[Merged by Bors] - feat: Port Algebra.Quandle

### PR DESCRIPTION
**Pull request by zaxioms0**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Quandle.lean [hunk 1]
```diff
@@ -0,0 +1,766 @@
...
+* `Rack.EnvelGroup` is left adjoint to `Quandle.Conj` (`toEnvelGroup.map`).
+  The universality statements are `toEnvelGroup.univ` and `toEnvelGroup.univ_uniq`.
+
+## Notation
+
+The following notation is localized in `quandles`:
+
+* `x ◃ y` is `shelf.act x y`
+* `x ◃⁻¹ y` is `rack.inv_act x y`
+* `S →◃ S'` is `shelf_hom S S'`
```
Comment (jcommelin) [inline Mathlib/Algebra/Quandle.lean:65]:
```suggestion
* `x ◃ y` is `Shelf.act x y`
* `x ◃⁻¹ y` is `Rack.inv_act x y`
* `S →◃ S'` is `ShelfHom S S'`
```

File: Mathlib/Algebra/Quandle.lean [hunk 2]
```diff
@@ -310,10 +316,11 @@ variable {S₁ : Type _} {S₂ : Type _} {S₃ : Type _} [Shelf S₁] [Shelf S
...
 
-@[simp]
-theorem toFun_eq_coe (f : S₁ →◃ S₂) : f.toFun = f :=
-  rfl
-#align shelf_hom.to_fun_eq_coe ShelfHom.toFun_eq_coe
+-- Porting Note: Syntactically equal in Lean4
+-- @[simp]
+-- theorem toFun_eq_coe (f : S₁ →◃ S₂) : f.toFun = f :=
+--   rfl
+-- #align shelf_hom.to_fun_eq_coe ShelfHom.toFun_eq_coe
```
Comment (jcommelin) [inline Mathlib/Algebra/Quandle.lean:323]:
```suggestion
#noalign
```

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Comment by zaxioms** [comment]
@jcommelin sorry I didn't noitce you already started the merge. I realized there was one thing I thought should be changed, but I can change it back if it was better before

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**